### PR TITLE
use SM dest duty station for calculating move cost: 169655057

### PR DESCRIPTION
**Pull request by lynzt**
## Description

Use SM destination duty station postal code for when calculating the cost of a move. 
(currently we use the SM's personal address)

## Reviewer Notes

When check ppm record and make sure it equals the destination duty station. Per product: we're not changing anything on the front end, so the use can enter what ever they want and the ppm.destinationPostalCode should be the destination duty station.

Splitting out check for short-haul move into separate pr.

## Setup

`make run_server`
`make run_client`
Enter orders and ppm info and check db for correct postal code. 

## Code Review Verification Steps

* [ ] Request review from a member of a different team.
* [ ] Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/169655057) for this change

### CODE REVIEW

File: pkg/handlers/internalapi/personally_procured_move.go [hunk 1]
```diff
@@ -92,14 +92,25 @@ func (h CreatePersonallyProcuredMoveHandler) Handle(params ppmop.CreatePersonall
 		advance = &a
 	}
 
+	order, err := models.FetchOrder(h.DB(), move.OrdersID)
```
Comment (Ryan-Koch) [inline pkg/handlers/internalapi/personally_procured_move.go]: This fetch might be an opportunity to use the QueryBuilder SO, if you think that makes sense also.

File: pkg/handlers/internalapi/personally_procured_move.go [hunk 2]
```diff
@@ -349,13 +356,14 @@ func (h UpdatePersonallyProcuredMoveEstimateHandler) updateEstimates(ppm *models
 	}
 
 	originDutyStationZip := ppm.Move.Orders.ServiceMember.DutyStation.Address.PostalCode
+	destinationDutyStationZip := ppm.Move.Orders.NewDutyStation.Address.PostalCode
```
Comment (LeDeep) [inline pkg/handlers/internalapi/personally_procured_move.go]: the handlers for `personally_procured_move_estimate` and `personally_procured_move_incentive` also call `ComputeLowestCostPPMove`, so the destination zip that gets used in those handlers will need to be updated to the new duty station zip.

File: pkg/handlers/internalapi/personally_procured_move.go [hunk 3]
```diff
@@ -349,13 +356,14 @@ func (h UpdatePersonallyProcuredMoveEstimateHandler) updateEstimates(ppm *models
...
 
-	distanceMilesFromOriginPickupZip, err := h.Planner().Zip5TransitDistance(*ppm.PickupPostalCode, *ppm.DestinationPostalCode)
+	distanceMilesFromOriginPickupZip, err := h.Planner().Zip5TransitDistance(*ppm.PickupPostalCode, destinationDutyStationZip)
```
Comment (LeDeep) [inline pkg/handlers/internalapi/personally_procured_move.go]: a similar function will need to be updated to use the new duty station zip in the handlers for `personally_procured_move_estimate` and `personally_procured_move_incentive`

File: pkg/handlers/internalapi/personally_procured_move_shared.go [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+	"github.com/gofrs/uuid"
+
+	"github.com/transcom/mymove/pkg/models"
+	"github.com/transcom/mymove/pkg/services"
+	"github.com/transcom/mymove/pkg/services/pagination"
+	"github.com/transcom/mymove/pkg/services/query"
+)
+
+// GetDestionDutyStationPostalCode returns the postal code associated with orders->new_duty_station->address
+func GetDestionDutyStationPostalCode(db *pop.Connection, ordersID uuid.UUID) (string, error) {
```
Comment (LeDeep) [inline pkg/handlers/internalapi/personally_procured_move_shared.go]: typo...`GetDestion...` should be `GetDestination...`

**Pull request merged**